packages/lib/src/vis/hooks.ts

```diff
@@ -83,22 +83,30 @@ export function useWheelCapture() {
   useEventListener(domElement, 'wheel', onWheel, { passive: false });
 }
 
-export function useCSSCustomProperties(...names: string[]): {
-  colors: string[];
-  refCallback: RefCallback<HTMLElement>;
-} {
+export function useCustomColors(
+  properties: Record<`--h5w-${string}`, string | string[]>
+): [string[], RefCallback<HTMLElement>] {
   const [styles, setStyles] = useState<CSSStyleDeclaration>();
+  const isDark = useMediaQuery('(prefers-color-scheme: dark)');
 
   // https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node
   const refCallback: RefCallback<HTMLElement> = useCallback(
     (elem) => setStyles(elem ? window.getComputedStyle(elem) : undefined),
     [setStyles]
   );
 
-  return {
-    colors: names.map((name) => {
-      return styles ? styles.getPropertyValue(name).trim() : 'transparent';
-    }),
-    refCallback,
-  };
+  if (!styles) {
+    // Return `transparent` colors on initial render
+    return [Object.keys(properties).map(() => 'transparent'), refCallback];
+  }
+
+  const colors = Object.entries(properties).map(([name, defaultColors]) => {
+    const [light, dark] = Array.isArray(defaultColors)
+      ? defaultColors
+      : [defaultColors, defaultColors];
+
+    return styles.getPropertyValue(name).trim() || (isDark ? dark : light);
+  });
+
+  return [colors, refCallback];
```

Comment:
What is the added value of this utility compared to our previous use of `useCSSCustomProperties` with defaults set in `LineVis` ?

If it is the change of default colors depending on the light/dark color scheme, then this can be done by changing the defaults in `LineVis` rather than encapsulating in an utility ?

Reply:
One problem was that `useCSSCustomProperties` was returning empty strings, which I thought was confusing as a way to indicate that the properties were not set. It was also forcing us to fall back to default values on separate lines instead of in the destructuring (since `''` is defined).

I also saw an opportunity to be more declarative, more like `var(--h5w-my-var, myFallback)`. With dark mode in the picture, I think this is all the more important. Detecting dark mode directly in `LineVis` would add unnecessary clutter.

But I can clean things up a bit. Hang on a sec.